### PR TITLE
Clean Up Documentation

### PR DESCRIPTION
**Pull request by mdmathias**
This PR performs just a bit of inline documentation clean up in `JSONDecodable.swift` and `JSONSubscripting.swift`.

We should also update [the documentation](http://bignerdranch.github.io/Freddy/) when this is merged.

### CODE REVIEW

File: Sources/JSONSubscripting.swift [hunk 1]
```diff
@@ -523,7 +525,7 @@ extension JSON {
     /// Retrieves a `[String: JSON]` from a path into JSON or a fallback if not
     /// found.
     /// - parameter path: 0 or more `String` or `Int` that subscript the `JSON`
-    /// - parameter fallback: Value to use when one is missing at the subscript
+    /// - parameter or: Fallback `Dictionary` to use when one is missing at the subscript.
```
Comment (zwaldowski) [inline Sources/JSONSubscripting.swift]: These aren't right. The `- parameter` tag refers to the internal argument name. [See also](https://swift.org/documentation/api-design-guidelines/#parameter-names).

Comment (mdmathias) [inline Sources/JSONSubscripting.swift]: Really?  I am surprised that a caller would want documentation on the internal parameter. That feels confusing to me.

Comment (zwaldowski) [inline Sources/JSONSubscripting.swift]: "External vs. internal" muddies what's going on. The external name is what you're doing with the argument. The internal name is the reasonable/canonical name for the type.

This serves as an example. Naming the parameter `or` doesn't make sense, it identifiers what is being done to the parameter `fallback`. `fallback` describes what the `or` is being done to. 

From the document I link:

> Choose these names to make documentation easy to read. For example, these names make documentation read naturally:
>
> ```swift
/// Return an `Array` containing the elements of `self`
/// that satisfy `predicate`.
func filter(_ predicate: (Element) -> Bool) -> [Generator.Element]
>
/// Replace the given `subRange` of elements with `newElements`.
mutating func replaceRange(_ subRange: Range, with newElements: [E])
```
>
> These, however, make the documentation awkward and ungrammatical:
>
> ```swift
/// Return an `Array` containing the elements of `self`
/// that satisfy `includedInResult`.
func filter(_ includedInResult: (Element) -> Bool) -> [Generator.Element]
> 
/// Replace the range of elements indicated by `r` with
/// the contents of `with`.
mutating func replaceRange(_ r: Range, with: [E])
```

Comment (mdmathias) [inline Sources/JSONSubscripting.swift]: 👍 Thanks, I see the point now.  It also looks like the standard library behaves this way as well.  I believe that I've made all of the needed changes.

**Pull request merged**